Homework/Homework Remedial/Week 3.py

```diff
@@ -0,0 +1,129 @@
+# Question 1
+def processing_queue(input_file):
+    queue = []
+    with open(input_file, 'r') as file:
+        for line in file:
+            command, name = line.strip().split(' ')
+            if command == "JUMP":
+                queue.insert(0, name)
+            elif command == "JOIN":
+                queue.append(name)
+
+    return queue
```

Comment:
Homework 3, Question 1 marks: 10/15.

Well done. However, this solution would be made much more efficient by using the "deque" data structure from the "collections" module, along with its methods "append" and "appendLeft".

Reply:
The code for the different parts of this question should also have been have separated into different files, to ease running them and testing the functionality. 
The file naming can also be improved. Please do not include spaces in file names. Instead, use underscores to join the different words in the file names. This makes the file easier to run in the terminal.